File: app/src/main/java/io/github/lazymockbean/utils/Annotations.java
package io.github.lazymockbean.utils;

import io.github.lazymockbean.annotation.LazyMockBean;
import io.github.lazymockbean.annotation.LazySpyBean;
import lombok.experimental.UtilityClass;

import java.lang.reflect.Field;

@UtilityClass
public class Annotations {
    public Boolean hasLazyMockBeanAnotations(Field field) {
        return field.isAnnotationPresent(LazyMockBean.class) || field.isAnnotationPresent(LazySpyBean.class);
    }

    public Class<?>[] getInjectTargets(Field field) {
        if (!hasLazyMockBeanAnotations(field)) {
            return new Class[0];
        }

        if (field.isAnnotationPresent(LazySpyBean.class)) {
            return field.getAnnotation(LazySpyBean.class).value();
        }
        return field.getAnnotation(LazyMockBean.class).value();
    }
}

File: app/src/main/kotlin/io/github/taesk/lazymockbean/utils/Annotations.kt
package io.github.taesk.lazymockbean.utils

import io.github.taesk.lazymockbean.annotation.LazyMockBean
import io.github.taesk.lazymockbean.annotation.LazySpyBean
import java.lang.reflect.Field
import kotlin.reflect.KClass

object Annotations {
    fun hasLazyMockBeanAnotations(field: Field): Boolean {
        return field.isAnnotationPresent(LazyMockBean::class.java) || field.isAnnotationPresent(LazySpyBean::class.java)
    }

    fun getInjectTargets(field: Field): Array<KClass<*>> {
        if (hasLazyMockBeanAnotations(field)) {
            return emptyArray()
        }

        if (field.isAnnotationPresent(LazySpyBean::class.java)) {
            return field.getAnnotation(LazySpyBean::class.java).value
        }
        return field.getAnnotation(LazyMockBean::class.java).value
    }
}
